src/lib/config/ModelConfig.cc

```diff
@@ -0,0 +1,1007 @@
+      std::cerr << "[SimEng:ModelConfig] Invalid ISA value of \"" << ISA
+                << "\" passed in config file due to \"" << result.message
+                << "\" error. Cannot continue with config validation. Exiting."
+                << std::endl;
+      exit(1);
+    }
+    // Set isa_
+    if (ISA == "AArch64") {
+      isa_ = ISA::AArch64;
+    } else if ("rv64") {
+      isa_ = ISA::RV64;
+    }
+  }
+  createGroupMapping();
+
+  expectations_["Core"].addChild(
+      ExpectationNode::createExpectation<std::string>("emulation",
+                                                      "Simulation-Mode"));
+  expectations_["Core"]["Simulation-Mode"].setValueSet(
+      std::vector<std::string>{"emulation", "inorderpipelined", "outoforder"});
+
+  const float clockFreqUpperBound = 10.f;
+  expectations_["Core"].addChild(
+      ExpectationNode::createExpectation<float>(1.f, "Clock-Frequency"));
+  expectations_["Core"]["Clock-Frequency"].setValueBounds(0.001f,
+                                                          clockFreqUpperBound);
+
+  // Early check on ["Core"]["Clock-Frequency"] as values are needed to inform
+  // the expected lower bound of the ["Core"]["Timer-Frequency"] value
+  uint64_t tFreqUpperBound = clockFreqUpperBound * 1000;
+  if (!isDefault) {
+    ValidationResult result =
+        expectations_["Core"]["Clock-Frequency"].validateConfigNode(
+            configTree_["Core"]["Clock-Frequency"]);
+    float clockFreq = configTree_["Core"]["Clock-Frequency"].as<float>();
+    if (!result.valid) {
+      std::cerr << "[SimEng:ModelConfig] Invalid Clock-Frequency value of \""
+                << clockFreq << "\" passed in config file due to \""
+                << result.message
+                << "\" error. Cannot continue with config validation. Exiting."
+                << std::endl;
+      exit(1);
+    }
+
+    tFreqUpperBound = clockFreq * 1000;
+  }
+
+  expectations_["Core"].addChild(
+      ExpectationNode::createExpectation<uint32_t>(100, "Timer-Frequency"));
+  expectations_["Core"]["Timer-Frequency"].setValueBounds<uint32_t>(
+      1, tFreqUpperBound);
+
+  expectations_["Core"].addChild(ExpectationNode::createExpectation<bool>(
+      false, "Micro-Operations", true));
+  expectations_["Core"]["Micro-Operations"].setValueSet(
+      std::vector{false, true});
+
+  if (isa_ == ISA::AArch64) {
+    expectations_["Core"].addChild(ExpectationNode::createExpectation<uint64_t>(
+        128, "Vector-Length", true));
+    expectations_["Core"]["Vector-Length"].setValueSet(
+        std::vector<uint64_t>{128, 256, 384, 512, 640, 768, 896, 1024, 1152,
+                              1280, 1408, 1536, 1664, 1792, 1920, 2048});
+
+    expectations_["Core"].addChild(ExpectationNode::createExpectation<uint64_t>(
+        128, "Streaming-Vector-Length", true));
+    expectations_["Core"]["Streaming-Vector-Length"].setValueSet(
+        std::vector<uint64_t>{128, 256, 384, 512, 1024, 2048});
```

Comment:
384 is not a valid SME Streaming Vector length.

Reply:
Good spot, accidentally left in